In [105]:
import pandas as pd

In [106]:
employee = pd.read_csv('employee.csv')

In [107]:
employee.head()

,Employee ID,Name,Department,Salary,Age,Experience
0,1,John,IT,120000.0,28.0,4.0
1,2,Sarah,HR,85000.0,32.0,7.0
2,3,Mike,IT,NaN,29.0,5.0
3,4,Emma,Finance,150000.0,41.0,15.0
4,5,David,HR,90000.0,NaN,6.0


# Explore the Dataset

In [108]:
employee.shape

(10, 6)

In [109]:
employee.columns

Index(['Employee ID', 'Name', 'Department', 'Salary', 'Age', 'Experience'], dtype='str')

In [110]:
employee.dtypes

Employee ID      int64
Name               str
Department         str
Salary         float64
Age            float64
Experience     float64
dtype: object

In [111]:
employee.info()

<class 'pandas.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Employee ID  10 non-null     int64  
 1   Name         10 non-null     str    
 2   Department   10 non-null     str    
 3   Salary       8 non-null      float64
 4   Age          9 non-null      float64
 5   Experience   9 non-null      float64
dtypes: float64(3), int64(1), str(2)
memory usage: 612.0 bytes


# Find Missing Values

In [112]:
employee.isna()

,Employee ID,Name,Department,Salary,Age,Experience
0,False,False,False,False,False,False
1,False,False,False,False,False,False
2,False,False,False,True,False,False
3,False,False,False,False,False,False
4,False,False,False,False,True,False
5,False,False,False,False,False,True
6,False,False,False,False,False,False
7,False,False,False,False,False,False
8,False,False,False,True,False,False
9,False,False,False,False,False,False


In [113]:
employee.isna().sum()

Employee ID    0
Name           0
Department     0
Salary         2
Age            1
Experience     1
dtype: int64

# Handle Missing Values

In [114]:
employee['Salary']

0    120000.0
1     85000.0
2         NaN
3    150000.0
4     90000.0
5    135000.0
6    145000.0
7     80000.0
8         NaN
9    125000.0
Name: Salary, dtype: float64

In [115]:
employee['Salary'] = employee['Salary'].fillna(employee['Salary'].mean())

In [116]:
employee

,Employee ID,Name,Department,Salary,Age,Experience
0,1,John,IT,120000.0,28.0,4.0
1,2,Sarah,HR,85000.0,32.0,7.0
2,3,Mike,IT,116250.0,29.0,5.0
3,4,Emma,Finance,150000.0,41.0,15.0
4,5,David,HR,90000.0,NaN,6.0
5,6,Lisa,IT,135000.0,35.0,NaN
6,7,Tom,Finance,145000.0,38.0,12.0
7,8,Anna,Marketing,80000.0,27.0,3.0
8,9,Chris,Marketing,116250.0,30.0,5.0
9,10,Sophia,IT,125000.0,31.0,8.0


In [117]:
employee['Age'] = employee['Age'].fillna(employee['Age'].mean())

In [118]:
employee['Experience']= employee['Experience'].fillna(employee['Experience'].mean())

In [119]:
employee.isna().sum()

Employee ID    0
Name           0
Department     0
Salary         0
Age            0
Experience     0
dtype: int64

# Salary Analysis

In [120]:
employee[['Name','Salary']].sort_values(by='Salary',ascending=False)

,Name,Salary
3,Emma,150000.0
6,Tom,145000.0
5,Lisa,135000.0
9,Sophia,125000.0
0,John,120000.0
8,Chris,116250.0
2,Mike,116250.0
4,David,90000.0
1,Sarah,85000.0
7,Anna,80000.0


In [121]:
employee['Salary'].mean()

np.float64(116250.0)

In [122]:
employee['Salary'].max()

np.float64(150000.0)

In [123]:
employee['Salary'].min()

np.float64(80000.0)

- Highest salary = 150000.0
- Lowest salary = 80000.0
- Average salary = 116250.0

# Department Analysis

In [124]:
employee.groupby('Department')['Name'].count()

Department
Finance      2
HR           2
IT           4
Marketing    2
Name: Name, dtype: int64

In [125]:
employee

,Employee ID,Name,Department,Salary,Age,Experience
0,1,John,IT,120000.0,28.000000,4.000000
1,2,Sarah,HR,85000.0,32.000000,7.000000
2,3,Mike,IT,116250.0,29.000000,5.000000
3,4,Emma,Finance,150000.0,41.000000,15.000000
4,5,David,HR,90000.0,32.333333,6.000000
5,6,Lisa,IT,135000.0,35.000000,7.222222
6,7,Tom,Finance,145000.0,38.000000,12.000000
7,8,Anna,Marketing,80000.0,27.000000,3.000000
8,9,Chris,Marketing,116250.0,30.000000,5.000000
9,10,Sophia,IT,125000.0,31.000000,8.000000


In [126]:
employee.groupby('Department')['Salary'].mean()

Department
Finance      147500.0
HR            87500.0
IT           124062.5
Marketing     98125.0
Name: Salary, dtype: float64

# Highest Paid Employees


In [127]:
employee[['Name','Salary']].sort_values(by='Salary',ascending=False).head(5)

,Name,Salary
3,Emma,150000.0
6,Tom,145000.0
5,Lisa,135000.0
9,Sophia,125000.0
0,John,120000.0


# Age Analysis

In [128]:
employee[['Name','Age']].sort_values(by='Age',ascending=False)

,Name,Age
3,Emma,41.000000
6,Tom,38.000000
5,Lisa,35.000000
4,David,32.333333
1,Sarah,32.000000
9,Sophia,31.000000
8,Chris,30.000000
2,Mike,29.000000
0,John,28.000000
7,Anna,27.000000


In [129]:
employee[['Age']].mean()

Age    32.333333
dtype: float64

In [130]:
employee.loc[employee['Age'].idxmax(),['Name','Age']]

Name    Emma
Age     41.0
Name: 3, dtype: object

In [131]:
employee.loc[employee['Age'].idxmin(),['Name','Age']]

Name    Anna
Age     27.0
Name: 7, dtype: object

- Youngest employee = Anna
- Oldest employee = Emma
- Average age = 32.333333

# Create Experience Levels

```bash
# Example
bios['height_category'] = bios['height_cm'].apply(lambda x: 'Short' if x < 165 else ('Average' if x < 185 else 'Tall')) 
```

In [132]:
employee['Experience_levels'] = employee['Experience'].apply(lambda x: 'Junior' if x < 5 else ('Mid' if x < 10 else 'Senior'))

In [133]:
employee

,Employee ID,Name,Department,Salary,Age,Experience,Experience_levels
0,1,John,IT,120000.0,28.000000,4.000000,Junior
1,2,Sarah,HR,85000.0,32.000000,7.000000,Mid
2,3,Mike,IT,116250.0,29.000000,5.000000,Mid
3,4,Emma,Finance,150000.0,41.000000,15.000000,Senior
4,5,David,HR,90000.0,32.333333,6.000000,Mid
5,6,Lisa,IT,135000.0,35.000000,7.222222,Mid
6,7,Tom,Finance,145000.0,38.000000,12.000000,Senior
7,8,Anna,Marketing,80000.0,27.000000,3.000000,Junior
8,9,Chris,Marketing,116250.0,30.000000,5.000000,Mid
9,10,Sophia,IT,125000.0,31.000000,8.000000,Mid


# Department Salary Report

In [134]:
report = employee.groupby("Department").agg({
    "Salary":["mean","max","min"],
    "Employee ID":"count"
})

# Export Report

In [135]:
report.to_csv(
    "department_salary_report.csv"
)